### DISCUSSION TITLE
a Cooler with non integer "counts"

### DISCUSSION THREAD

**Discussion by thekswenson** [open2c/cooler#434] [Q&A] [answered]
Hello,

Is it possible to store non-integer counts?
It looks like the Cooler design may only allow weights to be stored for bins, which allow for non-integer values when they are multiplied by the counts, but I'd like to store floats directly in the matrix.

I have good reasons to do this, as I do statistical tests that compare to a null hypothesis requiring a "randomized" matrix: I need to do this randomization on the normalized matrices.

Thanks.
(I didn't find a link to a mailing list, so I've posted the question here)

**Comment by nvictus** [comment, marked as answer]
You can definitely store non-integer counts. 

* API: in [create_cooler](https://cooler.readthedocs.io/en/latest/api.html#cooler.create_cooler), you can specify a dtype for the "count" field with `dtype={"count": float}`
* CLI: in [cooler load](https://cooler.readthedocs.io/en/latest/api.html#cooler.create_cooler) and [cooler cload](https://cooler.readthedocs.io/en/latest/cli.html#cooler-cload), there is a `--field` option that does something similar, and the former also has a shortcut flag `--count-as-float`

**Comment by thekswenson** [comment]
Great... thank you!
